tsinfer/formats.py

```diff
@@ -2004,33 +2058,92 @@ def num_alleles(self, sites=None):
                 num_alleles[j] -= 1
         return num_alleles[sites]
 
-    def variants(self, sites=None):
+    def variants(self, sites=None, recode_ancestral=None):
         """
         Returns an iterator over the Variant objects. This is equivalent to
-        the TreeSequence.variants iterator.
-        """
+        the TreeSequence.variants iterator. The alleles are guaranteed to be
+        returned in an order such that the ancestral state (if know) is the
+        first item in the list (and hence a ``0`` in the genotypes array) corresponds
+        to a sample with the ancestral state. This may mean that the indexes in the
+        genotypes array are recoded compared to the array passed in to :meth:`add_site`.
+
+        :param array sites: A numpy array of ascending site ids for which to return
+            data. If None (default) return all sites.
+        :param bool recode_ancestral: Should we recode the genotypes so that the
```

Comment:
i.e., "If True recode genotypes so that the ancestral state is coded as 0 as described above. Otherwise return genotypes in the input allele encoding"